cmd/gaia/app/app.go

```diff
@@ -26,16 +26,46 @@ import (
 	"github.com/tendermint/tendermint/libs/log"
 )
 
-const (
-	appName = "GaiaApp"
-)
+const appName = "GaiaApp"
 
-// default home directories for expected binaries
 var (
-	DefaultCLIHome  = os.ExpandEnv("$HOME/.gaiacli")
+	// default home directories for gaiacli
+	DefaultCLIHome = os.ExpandEnv("$HOME/.gaiacli")
+
+	// default home directories for gaiad
 	DefaultNodeHome = os.ExpandEnv("$HOME/.gaiad")
+
+	// The ModuleBasicManager is in charge of setting up basic,
+	// non-dependant module elements, such as codec registration
+	// and genesis verification.
+	ModuleBasics sdk.ModuleBasicManager
 )
 
+func init() {
+	ModuleBasics = sdk.NewModuleBasicManager(
+		genaccounts.AppModuleBasic{},
+		genutil.AppModuleBasic{},
+		auth.AppModuleBasic{},
+		bank.AppModuleBasic{},
+		staking.AppModuleBasic{},
+		mint.AppModuleBasic{},
+		distr.AppModuleBasic{},
+		gov.AppModuleBasic{},
+		params.AppModuleBasic{},
+		crisis.AppModuleBasic{},
+		slashing.AppModuleBasic{},
+	)
+}
+
+// custom tx codec
+func MakeCodec() *codec.Codec {
+	var cdc = codec.New()
+	ModuleBasics.RegisterCodec(cdc)
```

Comment:
Do we want `ModuleBasics` to be public? Anything that imports app now has a fully global codec. Which may be fine all together.

Reply:
yeah it's fine, we also need this to be exposed for usage by `gaiad` (and something else additionally)